### ISSUE TITLE
Disconnecting Minions

### ISSUE THREAD

**Issue by Silvenga** [saltstack/salt#25202]
With following Minion configuration:

```
id: minionName

master:
  - salt.<retracted>.com
  - salt.<retracted>.lan

master_type: failover

backup_mode: minion

rejected_retry: true
restart_on_error: true
auth_safemode: false
```

I'm seeing minions randomly disconnect from the Master until I manually restart them. I always see the following message spammed in the logs:

```
2015-07-07 06:47:52,549 [salt.utils       ][ERROR   ][1248] DNS lookup of 's' failed.
2015-07-07 06:47:52,550 [salt.minion      ][ERROR   ][1248] Master hostname: 's' not found. Retrying in 30 seconds
``` 

I have no idea where this hostname "s" is coming from. 

I can still run commands like:

```
salt-call test.ping
local:
    True
```

However, any communication from the master times out:

```
salt minionName test.ping
minionName:
    Minion did not return. [No response]
```

Salt Version:

```
salt-minion --versions-report
                  Salt: 2015.5.2
                Python: 2.7.6 (default, Jun 22 2015, 18:00:18)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
 Debian source package: 2015.5.2+ds-1trusty1
```

**Comment by sjmh** [comment]
See issue #24434 for what the 's' is.

Basically, the failover logic was broken by a PR.  The 's' is coming from the fact that it's accidentally converting the master hostname from a string into a list ['s','a','l','t'.. ].

**Comment by jfindlay** [comment]
@Silvenga, thanks for the report.  This issue should be fixed in 2015.5.3, which will come out in the next few days.  If you are still having this problem with that release, comment on #24434 and we'll reopen the issue.

**Comment by Silvenga** [comment]
@jfindlay you guys are quick. Thanks!
